backend/API_standards.md

```diff
@@ -0,0 +1,136 @@
+# API Standards
+
+## Naming conventions
+
+### PATH/Routes naming conventions
+All URL will follow `lisp-case` convention,
+
+### API input/output naming conventions
+All JSON objects and parameters sent and received from the API will follow the `snake_case` convention.
+
+Example:
+```json
+{
+  "first_name": "Jon",
+  "last_name": "Doe",
+  "age": 45
+}
+```
+
+Query string parameters will follow the same convention, example: `GET /api/questions?page_size=8&page_number=4`
+
+## API Requests
+
+### CRUD, Verbs, and Actions
+A resourceful route provides a mapping between HTTP verbs and URLs to
+controller actions. By convention, each action also maps to a specific CRUD
+operation in a database.
+
+Example for a `Question` resource:
+
+| HTTP Verb | Path             | Service#Function     | Used for                                     |
+| --------- | ---------------- | -------------------- | -------------------------------------------- |
+| GET       | /questions       | questions#index      | list all questions (filterable query string) |
+| POST      | /questions       | questions#create     | create a new question                        |
+| GET       | /questions/:id   | questions#show       | display a specific question                  |
+| PUT       | /questions/:id   | questions#update     | update a specific question                   |
+| DELETE    | /questions/:id   | questions#delete     | delete a specific question                   |
+
+Non CRUD operation will follow the same pattern, example:
+
+| HTTP Verb | Path              | Service#Function     | Used for                                     |
+| --------- | ----------------  | -------------------- | -------------------------------------------- |
+| GET       | /questions/search | questions#search     | search questions (filterable query string)   |
+
+Nested resources will follow the same pattern:
+
+| HTTP Verb | Path                 | Service#Function     | Used for                                     |
+| --------- | -------------------  | -------------------- | -------------------------------------------- |
+| GET       | /company/:id/members | members#index        | list all members for a specific company      |
+
```

Comment:
I kind of agree with both haha, as @lordaniel said sometimes resources without an identifier are needed, but as @scastillo said, `/me` is kind of weird.

For those cases, I believe we should be flexible and don't be so strict about REST, we can use a singularized form of the resource like `GET /user/questions` and `GET /user/answers`

For example, on Github API [GET /user/repos](https://developer.github.com/v3/repos/#list-your-repositories) gets the repositories owned by the authenticated user, the same pattern is followed by [GET /user/orgs](https://developer.github.com/v3/orgs/#list-your-organizations)

A similar approach is used at LinkedIn https://developer.linkedin.com/documents/profile-api#, tho I prefer the github one.